task_manager/statuses/views.py

```diff
@@ -65,7 +66,13 @@ def get(self, request: HttpRequest, **kwargs: int | str) -> HttpResponse:
     def post(self, request: HttpRequest, **kwargs: int | str) -> HttpResponse:
         status_id = kwargs.get("id")
         status = get_object_or_404(Status, pk=status_id)
-        status.delete()
-        message = _("Status has been deleted successfully")
-        messages.add_message(request, messages.SUCCESS, message)
+        tasks_having_status = Task.objects.filter(status=status)
+        # if status.task_set:
+        if tasks_having_status:
+            message = _("It is not possible to delete the status because it is in use")
+            messages.add_message(request, messages.ERROR, message)
+        else:
+            status.delete()
+            message = _("Status has been deleted successfully")
+            messages.add_message(request, messages.SUCCESS, message)
```

Comment:
wdyt about this option?
```suggestion
        tasks_with_this_status = Task.objects.filter(status=status)
        if tasks_with_this_status:
            message = _("It is not possible to delete the status because it is in use")
            messages.add_message(request, messages.ERROR, message)
        else:
            status.delete()
            message = _("Status has been deleted successfully")
            messages.add_message(request, messages.SUCCESS, message)
```